frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segment_statistics_modal.tsx

```diff
@@ -0,0 +1,173 @@
+  groupName: string;
+  volumeInNm3: number;
+  formattedSize: string;
+  volumeInVoxel: number;
+  boundingBoxTopLeft: Vector3;
+  boundingBoxPosition: Vector3;
+};
+
+const exportStatisticsToCSV = (
+  segmentInformation: Array<SegmentInfo>,
+  tracingId: string,
+  groupId: number,
+) => {
+  if (segmentInformation.length < 0) {
+    return;
+  }
+  const segmentStatisticsAsString = segmentInformation
+    .map(
+      (segmentInfo) =>
+        `${segmentInfo.segmentId},${segmentInfo.segmentName},${segmentInfo.groupId},${segmentInfo.groupName},${segmentInfo.volumeInVoxel},${segmentInfo.volumeInNm3},${segmentInfo.boundingBoxTopLeft},${segmentInfo.boundingBoxPosition}`,
+    )
+    .join("\n");
+  const csv = [SEGMENT_STATISTICS_CSV_HEADER, segmentStatisticsAsString].join("\n");
+  const filename = `segmentStatistics_tracing-${tracingId}_group-${groupId}.csv`;
+  const blob = new Blob([csv], {
+    type: "text/plain;charset=utf-8",
+  });
+  saveAs(blob, filename);
+};
+
+export function SegmentStatisticsModal({
+  isOpen,
+  onCancel,
+  tracingId,
+  tracingStoreUrl,
+  visibleSegmentationLayer,
+  relevantSegments: segments,
+  parentGroup,
+  groupTree,
+}: Props) {
+  const mag = getResolutionInfo(visibleSegmentationLayer.resolutions);
+  const nmFactorToUnit = new Map([[1, "nm³"]]);
+  const dataSource = useFetch(
+    async () => {
+      await new Promise((resolve) => setTimeout(resolve, 10000)); //TODO delete; only for testing
+      await api.tracing.save();
+      const segmentStatisticsObjects = await Promise.all([
+        getSegmentVolumes(
+          tracingStoreUrl,
+          tracingId,
+          mag.getHighestResolution(),
+          segments.map((segment) => segment.id),
+        ),
+        getSegmentBoundingBoxes(
+          tracingStoreUrl,
+          tracingId,
+          mag.getHighestResolution(),
+          segments.map((segment) => segment.id),
+        ),
+      ]).then((response) => {
+        const segmentSizes = response[0];
+        const boundingBoxes = response[1];
+        const statisticsObjects = [];
+        for (let i = 0; i < segments.length; i++) {
+          // segments in request and their statistics in the response are in the same order
+          const currentSegment = segments[i];
+          const currentBoundingBox = boundingBoxes[i];
+          const segmentStatObject = {
+            key: currentSegment.id,
+            segmentId: currentSegment.id,
+            segmentName:
+              currentSegment.name == null ? `Segment ${currentSegment.id}` : currentSegment.name,
+            groupId: currentSegment.groupId == null ? -1 : currentSegment.groupId,
+            groupName: getGroupNameForId(
+              currentSegment.groupId == null ? -1 : currentSegment.groupId,
+            ),
+            volumeInVoxel: segmentSizes[i],
+            volumeInNm3: parseInt(
+              formatNumberToUnit(segmentSizes[i], nmFactorToUnit).split(ThinSpace)[0],
+            ),
```

Comment:
this formats the number to an appropriate unit (which isn't necessarily nm³) and then strips that unit away and puts it into `volumeInNm3`? I think, that's wrong? :sweat_smile:
Also, the actual vx->nm conversion is missing (this isn't done by formatNumberToUnit). You need to multiply the size with the datasource.scale and then you are already done, I think.

Reply:
to be fair my `nmFactorToUnit` only consisted of nm^3, but of course you're right that I lacked understanding in the actual conversion 😅